iteration/iteration.go

```diff
@@ -26,19 +26,25 @@ const (
 	IterationStateClose    = "close"
 	PathSepInService       = "/"
 	PathSepInDatabase      = "."
+	IterationActive        = bool(true)
+	IterationNotActive     = bool(false)
 )
 
+var active bool
+
 // Iteration describes a single iteration
 type Iteration struct {
 	gormsupport.Lifecycle
-	ID          uuid.UUID `sql:"type:uuid default uuid_generate_v4()" gorm:"primary_key"` // This is the ID PK field
-	SpaceID     uuid.UUID `sql:"type:uuid"`
-	Path        path.Path
-	StartAt     *time.Time
-	EndAt       *time.Time
-	Name        string
-	Description *string
-	State       string // this tells if iteration is currently running or not
+	ID           uuid.UUID `sql:"type:uuid default uuid_generate_v4()" gorm:"primary_key"` // This is the ID PK field
+	SpaceID      uuid.UUID `sql:"type:uuid"`
+	Path         path.Path
+	StartAt      *time.Time
+	EndAt        *time.Time
+	Name         string
+	Description  *string
+	State        string // this tells if iteration is currently running or not
+	UserActive   *bool  // this can be nil
+	ActiveStatus bool
```

Comment:
Hi @baijum: one of the use cases was -
Say user logs into the system today and sees an iteration that has start-date of tomorrow. He does nothing with it.
Tomorrow, when he will open the system user should see that iteration in the active zone.
We can achieve this by doing computation on every `get iterations` request but I feel that it will be overhead for API, so we thought of having a scheduler or similar to do the job periodically. If that is not expensive enough to do in API then let's do it in API and we won't need that status in DB.

If we can achieve this without something like a scheduler then it is okay not to store that value in DB.